domain/consensus/processes/blockprocessor/blockprocessor.go

```diff
@@ -54,25 +70,40 @@ func New(
 		difficultyManager:     difficultyManager,
 		pastMedianTimeManager: pastMedianTimeManager,
 		ghostdagManager:       ghostdagManager,
+		coinbaseManager:       coinbaseManager,
 
 		consensusStateManager: consensusStateManager,
 		acceptanceDataStore:   acceptanceDataStore,
 		blockStore:            blockStore,
 		blockStatusStore:      blockStatusStore,
 		blockRelationStore:    blockRelationStore,
+		multisetStore:         multisetStore,
+		ghostdagDataStore:     ghostdagDataStore,
+		consensusStateStore:   consensusStateStore,
+		pruningStore:          pruningStore,
+		reachabilityDataStore: reachabilityDataStore,
+		utxoDiffStore:         utxoDiffStore,
 	}
 }
 
-// BuildBlock builds a block over the current state, with the transactions
-// selected by the given transactionSelector
+// BuildBlock builds a block over the current state, with the given
+// coinbaseData and the given transactions
 func (bp *blockProcessor) BuildBlock(coinbaseData *externalapi.DomainCoinbaseData,
 	transactions []*externalapi.DomainTransaction) (*externalapi.DomainBlock, error) {
 
-	return nil, nil
+	start := time.Now()
+	log.Debugf("BuildBlock start")
+	block, err := bp.buildBlock(coinbaseData, transactions)
+	log.Debugf("BuildBlock end. Took: %s", time.Since(start))
+	return block, err
 }
 
 // ValidateAndInsertBlock validates the given block and, if valid, applies it
 // to the current state
 func (bp *blockProcessor) ValidateAndInsertBlock(block *externalapi.DomainBlock) error {
-	return nil
+	start := time.Now()
```

Comment:
I suggest maybe have this thing infrastracturized.
a.k.a. Add an infra function 
`func logAndMeasureExecutionTime(funcName string) deferredCallback`
And then you'll call 
`defer logAndMeasureExecutionTime("ValidateAndInsertBlock")()` in the beginning of the function, and be finished with this.